SKYKitChat/Classes/UI/ConversationView/SKYChatConversationViewController.swift

```diff
@@ -0,0 +1,612 @@
+
+extension SKYChatConversationViewController {
+
+    open func fetchParticipants(completion: (([SKYRecord]?, Error?) -> Void)?) {
+        guard self.conversation != nil else {
+            print("Cannot fetch participants with nil conversation")
+            return
+        }
+
+        let participantIDs = self.conversation!.participantIds
+            .map { (eachParticipantID) -> SKYRecordID in
+                return SKYRecordID(recordType: "user", name: eachParticipantID)
+            }
+
+        self.skygear.publicCloudDatabase?
+            .fetchRecords(withIDs: participantIDs, completionHandler: { (result, error) in
+                guard error == nil else {
+                    print("Failed to fetch participants: \(error?.localizedDescription)")
+                    completion?(nil, error)
+                    return
+                }
+
+                guard let participantMap = result as? [SKYRecordID: SKYRecord] else {
+                    print("Fetched participants are in wrong format")
+                    let err = self.errorCreator.error(with: SKYErrorBadResponse,
+                                                      message: "Fetched participants are in wrong format")
+                    completion?(nil, err)
+                    return
+                }
+
+                var participants: [SKYRecord] = []
+                for (k, v) in participantMap {
+                    self.participants[k.recordName] = v
+                    participants.append(v)
+                }
+
+                if let senderRecord = self.participants[self.senderId],
+                    let senderName = senderRecord.object(forKey: "name") as? String {
+
+                    self.senderDisplayName = senderName
+                }
+
+                completion?(participants, nil)
+
+                self.collectionView?.reloadData()
+                self.collectionView?.layoutIfNeeded()
+
+            }, perRecordErrorHandler: nil)
+    }
+
+    open func fetchMessages(before: Date?, completion: (([SKYMessage]?, Error?) -> Void)?) {
+        guard self.conversation != nil else {
+            print("Cannot fetch messages with nil conversation")
+            return
+        }
+
+        let chatExt = self.skygear.chatExtension
+
+        chatExt?.fetchMessages(
+            conversation: self.conversation!,
+            limit: Int(self.messagesFetchLimit),
+            beforeTime: before,
+            completion: { (msgs, error) in
+                guard error == nil else {
+                    print("Failed to fetch messages: \(error?.localizedDescription)")
+                    completion?(nil, error)
+                    return
+                }
+
+                guard msgs != nil else {
+                    print("Failed to get any messages")
+                    let err = self.errorCreator.error(with: SKYErrorBadResponse,
+                                                      message: "Failed to get any messages")
```

Comment:
The default is plural, unless when you are specifically talking about one item.

https://en.wikipedia.org/wiki/English_plurals#Equivalent_to_zero_is_usually_plural
http://english.stackexchange.com/questions/112996/any-followed-by-singular-or-plural-countable-nouns